Many updates

nirgal opened the pull request:
If you wish, I have a full batch of updates, I merged from a lot of forks. You'll find mainly:

Project cleaning:
- project structure from code.google (not ideal yet), including init.d script
- README file
- python cleaning: 4 spaces indentation, lower case variable names
- explicit GPLv3 license.

Existing feature improvement:
- Optimize integrity digest computation, and do it on the fly. So now it works with big files without timing out.
- connection Keep-Alive/Close: One should not call handle_one_request from do_POST

New features
- support for multiple cards.
- Save decoded data during http transfer, on the fly. This might be used by a monitoring applet in the future.
- Resume file transfer after timeout, rather than transfering it again from scratch.
- Save to date related directory, based either on transfer date or file date from tar headers.
- Log transfer speed.
- Support SIGHUP to reload configuration without restart.
- Hook to execute a custom command on file completion.

That pull request was last updated 2012-09-04

tachang committed:
Resurected some original documentation files

tachang committed:
Initial commit

tachang committed:
Initial commit

tachang committed:
github generated gh-pages branch

tachang committed:
Updated transfermode so that it is no longer hardcoded.

tachang committed:
Update transfermode & timestamp, not hard coded

Version on google discarded theses changes.

tachang committed:
Updated timestamp so it is no longer hard coded

Commit:
This is a slight rework of a post of Jeff Tchang at Return Boolean True

jsgf committed:
Rearrange tree; delete cruft files.

jsgf committed:
Convert to Unix EOL

jsgf committed:
Do ~ expansion on picture directory name, and make any missing
intermediate directories.

jsgf committed:
Do ~ expansion on picture directory name

Commit:
+ ‘Release 2.0/EXIF.py’
+ ‘Release 2.0/EXIF.pyc’
+ ‘Release 2.0/EyeFiServer-2.1.1.py’
+ ‘Release 2.0/EyeFiServer.log’
+ ‘Release 2.0/TestOverwrite.py’
+ ‘Release 2.0/TestRW2.py’
+ ‘Release 2.0/changelog.rtf’

Commit:
/ ‘Release 2.0/EyeFiServer-2.1.1.py’

Commit:
/ ‘Release 2.0/DebugSettings.ini’
/ ‘Release 2.0/EXIF.pyc’
/ ‘Release 2.0/EyeFiCrypto.pyc’
/ ‘Release 2.0/EyeFiSOAPMessages.pyc’
/ ‘Release 2.0/EyeFiServer-2.1.1.py’
/ ‘Release 2.0/configobj.pyc’

Commit:
+ ‘Release 2.0/.~lock.changelog.rtf#’
/ ‘Release 2.0/EyeFiServer-2.1.1.py’
/ ‘Release 2.0/changelog.rtf’

Commit:
/ ‘Release 2.0/.~lock.changelog.rtf#’

Commit:
/ ‘Release 2.0/.~lock.changelog.rtf#’
/ ‘Release 2.0/changelog.rtf’

Commit:
/ ‘Release 2.0/.~lock.changelog.rtf#’
/ ‘Release 2.0/changelog.rtf’

Commit:
Add INTEGRITYDIGEST Verification

Commit:
Add support for multiple cards

nirgal committed:
Make pylint slightly more happy

nirgal committed:
Code cleanup, mostly docstrings

nirgal committed:
Added GPLv3+ license prefix

See https://github.com/tachang/EyeFiServer/issues/5

nirgal committed:
Moved code back in one file. INSTALL

This will easy installation procedure / packaging

nirgal committed:
Fixed misleading comment about empty hostname

It listen to * then, not to loopback

nirgal committed:
README file

nirgal committed:
KeepAlive, code factorisation, misc...

Fixed key in example eyefiserver.conf

Added note about threaded exception not beeing logged in logfile,
but only on stderr

Change date fomat. Use default, these AM/PM things doesn't mean much
in some countries.

Changes eyeFiLogger.debug parameters to use % rather than +

Proper way to analyse HTTP headers in python

Show transfer speed in debug log

Big change is the debuging of connection keep-alive/close:
In do_POST, on startSession, we where calling handle_one_request (!!)
This big hack was recursively calling rfile.read() and bypassing
"Connection:" headers.
Now we do return for do_POST, and make sure to issue "Keep-Alive"
header unless we close the connection.

nirgal committed:
Simplified XML SAX parser a lot

and fixed an wrong "except: e1, e2" where "," was understood as "as"

nirgal committed:
Use default for config hostname, hostport

People usually won't want to change it
So I remove it from example conf file, and add support for default
values

This extra options should be documented somewhere however.

Commit:
Execute a command after each file

Can be used to display on the fly, upload to the cloud, ...

nirgal committed:
Fixes in config reading

nirgal committed:
Drop support for python <2.6: Clean shutdown

nirgal committed:
Stop using bytes standard class name as a variable

nirgal committed:
Changed parameters into options, default values

nirgal committed:
Added socket timeout so that SD probes don't hang

nirgal committed:
Imported Upstream version 0.1+git20120823

nirgal committed:
First version

nirgal committed:
Switch from SafeConfigParser to RawConfigParser

So that % don't have to be escaped any more in the config files.

nirgal committed:
Add support for SIGHUP to reload configuration

Also drop support for GET querries, that were not fully implemented and
were only there to support the stop requests.

nirgal committed:
Drop support for server address/port

Rationale: We usually want to listen to external connection, so '' is a
good choice. One might want to filter to a specific interface for
filtering? This is a bit far fetched, and not usefull.
Regarding the port, it is allways 59278. Changing it has no meaning, but
for development.

nirgal committed:
Add option to customise loglevel in config file

nirgal committed:
Code factorisation: SOAP response builder

nirgal committed:
Aesthetic changes

Shorter lines
More docstrings
...

nirgal committed:
Normalized identation \t = 4 spaces

I think this is ithe usual convention
Fixes pylint W0311

nirgal committed:
Asthetic changes, code factorisation, log messages

nirgal committed:
Load data by chunk, print download progress

Meaning post data parsing before post is complete
Factorise some code
Also fixed calculateIntegrityDigest after 4 spaces intentation

nirgal committed:
Write tar file while it is received

This will allow thumbnail previews for a future applet

nirgal committed:
make pylint happier with shorter lines

nirgal committed:
Repport download progress every second

rather than every X bytes. This scales much better whatever the
bandwidth.

nirgal committed:
Make pylint happier

Renamed many vars to follow python guidelines.

nirgal committed:
Added option use_date_from_file for dir selection

This uses the date from the tar header
Default is disabled

nirgal committed:
Optimise (fix) calculate_tcp_checksum

Grouping the calls to struct() results in huge performance boost.
On my laptop, for a 6MB picture, integritydigest speed changed from 7"
to 2".
That function caused timeout in uploadPhoto resulting in the pic to be
sent over and over.

nirgal committed:
calculate_integritydigest: Removed 512 block pad

Adding '\0' does not change the sum of ushorts. So this is just useless.

nirgal committed:
Compute INTEGRITYDIGEST while file is writen

This fixes timeouts for big files

nirgal committed:
Don't keep posted data in memory but the last 2k

This reduce memory fingerprint for big movies.

nirgal committed:
Removed obsolete comments

xmlns="http://localhost/api/soap/eyefilm" should always be sent.

nirgal committed:
Add support for transfer resume after timeout

Rather than transering the whole file again.
This implies some changes:
- New EyeFiSession for keeping data accros requests
- IntegrityDigestFile seek() now actually read the file and updates the
  current digest
- Files are save in the data-independent part of upload_dir

nirgal committed:
Add GetPhotoStatus md5 challenge check

Moved all the md5 computation in EyeFiSession
"snonce" is now randomly generated rather than hard coded
EyeFiRequestHandler.send_eyefi_response sends proper 500 if there is no
response.

nirgal committed:
Moved config parsing to new class EyeFiConfig

Also replaced funcName by levelname in LOG_FORMAT

nirgal committed:
Don't log anything until after logvel is set

nirgal committed:
Force Connection:Close on errors

Means an EyeFiRequestHandler.__init__ method (see code comments for
bugs)

nirgal committed:
Abort file transfers on shutdown requests

This speed up shutdown() a lot obviously.

nirgal committed:
Added myself to auhors

nirgal committed:
Renamed eyefiserver.py to eyefiserver

When scripts are installed into a directory in the system PATH, the
script name should not include an extension such as .sh or .pl that
denotes the scripting language currently used to implement it.
-- Debian policy 10.4
It makes sense, doesn't it?

nirgal committed:
Default is now INTEGRITYDIGEST check turned on

nirgal committed:
Add man source from Debian repo

nirgal committed:
New eyefiserver.conf(5); manpages in README

nirgal committed:
Double the \ characters in txt2man input

tachang commented:
Wow that's a lot of stuff :) Have you tried running this code and it works? Don't have my Eye-Fi card around right now to test it.

nirgal committed:
Improved man pages

nirgal commented:
Of course I tried it! :p

I plan to make a few more changes, like adding a proper hook for progress feedback.
I tested locally that the partial tar file can be used by tarfile lib to extract the partial file, and from it extract the thumbnail... :)
Man pages needs to be improved a little bit
Example eyefiserver.conf contains a mix of = and : as a separator, this need to be unified.

In the debian package I work on - see http://anonscm.debian.org/gitweb/?p=collab-maint/eyefiserver.git, I plan not to start the server by default, as it runs fine as a normal user. This is not implemented yet. And init.d script is buggy (need to remove /local/ in Debian) and is not using lsb scripts...

If you merge the pull request, please update the README file with a proper official version number.

I guess I won't work on that project before a week, as I now have other thing to do...

Any chance you can put your hands on your Eye-Fi card soon?

nirgal committed:
Now using key=value everywhere

There were some key:value left in some places.

nirgal committed:
More documentation

Added manual pages in the README tree.
Added section and volume names during man page generation.
Tuned man style.

nirgal committed:
Doc: New high level session dump, on SOAP level

nirgal committed:
More documentation

Renamed protocol to README.developer
New doc/README.developer with explanations of the different files

nirgal committed:
Progress meter hook

Changed in config file:
"execute" hook is now renamed "complete_execute"
New hook "progress_execute" called during file transfer
Updated documentation

Also removed unneeded CFGFILE from etc/init.d/eyefiserver

nirgal commented:
I'm done with the changes I needed. :)

nirgal committed:
Updated project structure

Going to a more usual project structure:
* No exemple var/log/eyefiserver.log
* Move code in src/ folder

Updated README & .gitignore to reflect the changes

nirgal committed:
Fix getboolean default value

True/False doesn't have a .lower() method.

nirgal committed:
Improved error handling

- Exemple eyefiserver.conf doesn't have 0000000 upload_key exemple
so that server hints about eyefiserver.conf(5) when missing.
- Nicer messages about write permission denied in upload_dir

nirgal committed:
Fixed return code on permission denied

so that pic is retried.

nirgal committed:
Added magic modeline for some editors

nirgal committed:
Fixed spelling errors in man pages

nirgal committed:
Releasing as version 2.3 rc1

nirgal commented:
I tagged that revision as 2.3_rc1 in my fork
https://github.com/nirgal/EyeFiServer/tags

nirgal commented:
Unofficial Debian package (unstable) is now available at debexpo:

$ dget -ux http://mentors.debian.net/debian/pool/main/e/eyefiserver/eyefiserver_2.3~rc1-1.dsc
$ cd eyefiserver-2.3~rc1/
$ debuild

nirgal commented:
Sponsors have pointed out that keeping a 4MB http dump file is not smart, I'll shorten it in a release candidate 2 soon.

nirgal committed:
Remove picture data from dump_http and compress it

This greatly reduces the overall size of the project

nirgal committed:
Releasing as version 2.3 rc2

nirgal commented:
That version is now part of Debian:
http://packages.qa.debian.org/eyefiserver

johnseekins committed:
Added current directory to config path

Now we can also use the current directory for getting configs. Nice if you want to avoid polluting your /usr/ and /etc trees and use /opt or something.

johnseekins committed:
Switch to optparse

johnseekins committed:
first commit towards daemons!

Begin handling logging at varying levels based on how the tool is started.

johnseekins committed:
Fix default loglevel value

Commit:
Use getboolean() to read use_date_from_file config

Commit:
Add an option to correct modification times

nirgal committed:
Update copyrights

nirgal committed:
Add full license text

nirgal committed:
Typo in manual

nirgal committed:
Release version 2.3

nirgal committed:
Update README version

nirgal committed:
Update update-rc.d hint

nirgal commented:
Since this repository looks dead, with no activity for years, I'm taking the liberty to release a version 2.3.
https://github.com/nirgal/EyeFiServer/tags

tachang commented:
Feel free I haven't updated in ages :)

On Sat, Apr 4, 2015 at 9:17 PM, nirgal notifications@github.com wrote:

> Since this repository looks dead, with no activity for years, I'm taking
> the liberty to release a version 2.3.
> https://github.com/nirgal/EyeFiServer/tags
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/tachang/EyeFiServer/pull/6#issuecomment-89714126.

nirgal committed:
Fix in argparse

nirgal committed:
Don't send some headers twice

Date: & Server: are already sent automatically by BaseHTTPRequestHandler.send_response()

nirgal committed:
Move correct_mtime doc for exemple .conf to man

nirgal committed:
Work around buggy UTC correction in tar files

This fixes an issue when the file was saved at a date where day light saving was different from today.

nirgal committed:
isort the imports

nirgal committed:
Fix invalid %] in error message

nirgal committed:
Style fix to pass flake8 tests

This include PEP8 coding style checks.

nirgal committed:
Documentation

Add version in manual generation invocation (README).
Wrap manual source to 80 columns.
Fix quotations in eyefiserver.conf examples.
Use sh rather than bash in example.
argparse: Add program description, Add loglife metavar.
Add -h alternative in manual.

nirgal committed:
Add changelog

nirgal committed:
Made EyeFiSession a "new style" class

nirgal committed:
Cleanup loglevel setup checks

Trust setLevel to do the checks.
Don't use eval anymore.

nirgal committed:
Cleanup random nonce generation

nirgal committed:
Copy warning about unlogged exceptions in man

nirgal committed:
Update copyright years

nirgal committed:
Use python3 division. Use ints in progress meter.

nirgal committed:
Use python2 in shebank (PEP394)

nirgal committed:
Releasing version 2.4

nirgal committed:
Fixed spelling typo in manual

nirgal committed:
Spelling typo in manual

nirgal committed:
Spelling typo in manual
